contrastscan-remote/src/main/java/com/contrastsecurity/scan/PreScanDataGenerator.kt

```diff
@@ -0,0 +1,81 @@
+package com.contrastsecurity.scan
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.LoggerFactory
+import java.io.FileWriter
+import java.io.IOException
+import java.io.Writer
+import java.net.URI
+import java.nio.file.*
+import java.nio.file.attribute.BasicFileAttributes
+
+sealed class PreScanDataGenerator {
+    companion object {
+
+        @JvmStatic
+        private val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
+
+        @Throws(Exception::class)
+        fun generate(sourceDir: Path, outPath: Path) {
+            logger.debug("generating prescan data: source dir [$sourceDir],  output path [$outPath]")
+            val mapper = ObjectMapper()
+            val writer: Writer = FileWriter(outPath.toFile())
```

Comment:
idiomatic Kotlin would let type inference play out here and live with `writer` being of type FileWriter